Featured experiments

meandavejustice opened the pull request:
Opening this up to get some eyes on it while I'm finishing up a couple of tasks

- fixes #3003 
- fixes #3000 
- fixes #3002 
- fixes #2999 
- fixes #2998 
- fixes #3072
- fixes #3149 
- fixes #2696

### Running
In order to run, you can just add `is_featured:true` to one of the current experiments in `content-src` and `npm start`

### Todo
- [x] get rest of the new tests passing
- [x] finish implementing legal modal now that photon modals pr has been merged ( might take some slight reworking of the Modal component)
- [x] tweaks to featured experiment and news feed page for small device views
- [ ] figure out why the icons from ExperimentPlatform component still aren't white cc @johngruen (I tried updating this in inkscape and they appear white for me in there, maybe this is some sort of firefox svg rendering issue ?)

Commit:
setup structural html for featured experiments

Commit:
setup structural html for featured experiments

Commit:
add news feed page, more styling for featured experiment

- created newsFeedPage including boilerplate files for tests and stories
- updated header to include news feed link
- little more work on featured experiment css and layout

Commit:
add news feed page, more styling for featured experiment

- created newsFeedPage including boilerplate files for tests and stories
- updated header to include news feed link
- little more work on featured experiment css and layout

Commit:
Featured Experiment style/layout tweaks, privacy note, install btn

Commit:
Featured Experiment style/layout tweaks, privacy note, install btn

Commit:
more css updates, platforms tweaks and componentize Measurements
- pull Measurments section into it's own component
- fix experiment platform selector
- add white icons for platform selector in featured experiments
- more css tweaks to featured experiment and update list

Commit:
more css updates, platforms tweaks and componentize Measurements
- pull Measurments section into it's own component
- fix experiment platform selector
- add white icons for platform selector in featured experiments
- more css tweaks to featured experiment and update list

Commit:
manage/feedback buttons and news feed experiment icon colors

Commit:
manage/feedback buttons and news feed experiment icon colors

Commit:
set cookie when visiting the news feed page

Commit:
set cookie when visiting the news feed page

Commit:
add "new" flag to updates, tweak css, add active state to header link

Commit:
add "new" flag to updates, tweak css, add active state to header link

Commit:
add featured experiment selectors, yml prop ref etc..

Commit:
add featured experiment selectors, yml prop ref etc..

Commit:
minor cleanups for wip pr

Commit:
minor cleanups for wip pr

flodolo commented on locales/en-US/app.ftl:
```diff
@@ -85,6 +84,9 @@ emailOptInDialogErrorTitle = Oh no!
 showMoreNewsTitle = Show Past News
 
+[[featuredExperiment]]
+moreDetail = More Detail
```
Not l10n related, but this doesn't sound right (details)?

meandavejustice replied:
I think works both ways but whichever sounds most natural should go in. I just followed the spec without questioning it, but happy to pluralize it

Commit:
Add legal modal

Commit:
Add legal modal

MikkCZ committed:
Pontoon: Update Czech (cs) localization of Test Pilot Website

Localization authors:
- Michal Stanke <mstanke@mozilla.cz>

Commit:
wip add tour dialog to featured experiments

Commit:
wip add tour dialog to featured experiments

Commit:
Breaking featuredButton and featuredStatus out into components

Commit:
Breaking featuredButton and featuredStatus out into components

Commit:
Finish adding tour dialog to feature experiment component

Commit:
Finish adding tour dialog to feature experiment component

Commit:
Get tests to pass

Commit:
Get tests to pass

Commit:
Update mobile views, Add stories, Add tests, cleanup

Commit:
Update mobile views, Add stories, Add tests, cleanup

Commit:
remove excess experimentTourDialog from experimentpage folder (borked in rebase

Commit:
remove excess experimentTourDialog from experimentpage folder (borked in rebase

Commit:
modify newsFeedActive check in header for some tests

Commit:
modify newsFeedActive check in header for some tests

Commit:
add back failOnError option to watching linter

Commit:
add back failOnError option to watching linter

Commit:
remove extra image and commented CSP for dev

Commit:
remove extra image and commented CSP for dev

Commit:
remove accidental added dependency and package-lock

Commit:
remove accidental added dependency and package-lock

Commit:
fix flowtype errors

Commit:
fix flowtype errors

AG12r committed:
Pontoon: Update Georgian (ka) localization of Test Pilot Website

Localization authors:
- Georgianizator <georgianization@outlook.com>

AG12r committed:
Pontoon: Update Georgian (ka) localization of Test Pilot Website

Localization authors:
- Georgianizator <georgianization@outlook.com>

Commit:
fix .secondary button background after recent color changes

Commit:
fix .secondary button background after recent color changes

lmorchard commented on docs/content/template.yaml:
```diff
@@ -3,7 +3,7 @@ slug: experiment-name
 subtitle: 'Powered by a Yellow Submarine'
 order: 0
-
+is_featured: true
```
Might want this `false` by example in the template

lmorchard commented on frontend/src/app/components/ExperimentTourDialog/index.js:
```diff
@@ -54,17 +54,17 @@ export default class ExperimentTourDialog extends React.Component {
         <div key={idx} className="step-content">
           <div className="step-image">
-            <img src={step.image} width="580" height="332"/>
+            <img src={step.image} />
```
Looks like this change will cause some reflows & size jumps as things load. Are these images all different sizes? Could we find a way to keep a consistent size here?

johngruen replied:
images are all the same size....i dropped the w and h in in a previous fix. this should be added here.

lmorchard commented on frontend/src/app/components/FeaturedExperiment/FeaturedButton.js:
```diff
@@ -0,0 +1,136 @@
+            </Localized>);
+  }
+
+  handleManage() {
```
nit: Might be good to update these to use the class property syntax to get these all bound in the constructor.

So you can have `handleManage = () => { ... }` here and `onClick={this.handleManage}` in the JSX - rather than `onClick={() => this.handleManage()`

lmorchard commented on frontend/src/app/components/FeaturedExperiment/FeaturedStatus.js:
```diff
@@ -0,0 +1,77 @@
+    );
+  }
+
+  justUpdated() {
```
Are these bits of just updated / launched logic duplicated from the experiment page? Would be good to pull them out into a util lib to DRY things up

meandavejustice replied:
They are indeed, I wasn't quite sure where to put the `justUpdated` and `justLaunched` methods.

lmorchard commented on frontend/src/app/components/FeaturedExperiment/stories.js:
```diff
@@ -0,0 +1,119 @@
+  video_url: 'https://www.youtube.com/embed/wjvM18UWMmM'
+};
+
+const baseProps = {
```
Clicking on the install button in the first story, I get an `installAddon is not a function` error. Should add `installAddon: action('installAddon')` here to catch that - or even better, something that might simulate an install with a delay

lmorchard commented on locales/en-US/app.ftl:
```diff
@@ -32,18 +32,17 @@ landingIntroOne = Test new features.
 [[oneClickInstall]]
 oneClickInstallMinorCta = Install Test Pilot &amp;
 oneClickInstallMajorCta = Enable {$title}
+enabledExperimentCta =  Enable {$title}
```
Does this need to be a separately translated string? Or could we re-use `oneClickInstallMajorCta `

meandavejustice replied:
Should be fine to reuse. I'll fix that up

lmorchard reviewed (commented):
Did a quick skim through, came up with a few comments that I think are all just minor nits. Will do a deeper dive later tonight

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

ujdhesa committed:
Pontoon: Update Albanian (sq) localization of Test Pilot Website

Localization authors:
- Besnik Bleta <besnik@programeshqip.org>

selimsum committed:
Pontoon: Update Turkish (tr) localization of Test Pilot Website

Localization authors:
- Selim Şumlu <selim@sum.lu>

selimsum committed:
Pontoon: Update Turkish (tr) localization of Test Pilot Website

Localization authors:
- Selim Şumlu <selim@sum.lu>

selimsum committed:
Pontoon: Update Turkish (tr) localization of Test Pilot Website

Localization authors:
- Selim Şumlu <selim@sum.lu>
- Ali Demirtaş <alidemirtas94@gmail.com>

selimsum committed:
Pontoon: Update Turkish (tr) localization of Test Pilot Website

Localization authors:
- Selim Şumlu <selim@sum.lu>

Commit:
Pontoon: Update Norwegian Nynorsk (nn-NO) localization of Test Pilot Website

Localization authors:
- Bjørn I. <bjorn.svindseth@online.no>

Commit:
Pontoon: Update Norwegian Nynorsk (nn-NO) localization of Test Pilot Website

Localization authors:
- Bjørn I. <bjorn.svindseth@online.no>

Commit:
Address nits from pr review...

- pull justLaunched and justUpdated out into their own file to keep
things DRY.
- remove setExperiementLastSeen and getExperimentLastSeen and all
references.
- added more tests

Commit:
Address nits from pr review...

- pull justLaunched and justUpdated out into their own file to keep
things DRY.
- remove setExperiementLastSeen and getExperimentLastSeen and all
references.
- added more tests

Commit:
Address issue with enabled label on tour header

- refs #3149

Commit:
Address issue with enabled label on tour header

- refs #3149

lmorchard reviewed (approved):
Ran through this again and any nits I had are covered. Could use yet another pair of eyes to be thorough. Beyond that, review for UX and whatever tweaks come up from there would be good

pdehaan commented on docs/content/reference.md:
````diff
@@ -111,6 +111,14 @@ The order in which the experiment will appear. Lower numbers appear first. Requi
+Determines whether the experiment will be featured on the home page. Optional.
+
+```yaml
+order: 0
````
Should this be `order: 0` or `is_featured: true`, or \<other\>?

meandavejustice replied:
should be `is_featured`, I'll push a fix now

Commit:
update is_featured code snippet from reference.md

Commit:
update is_featured code snippet from reference.md

Commit:
update is_featured code snippet from reference.md

Commit:
update is_featured code snippet from reference.md

flodolo commented on locales/en-US/app.ftl:
```diff
@@ -146,7 +148,7 @@ tourLink = Launch Tour
 bugReports = Bug Reports
 discussExperiment = Discuss { $title }
-tourOnboardingTitle = {$title} enabled!
+tourOnboardingTitleGeneral = {$title}
```
Since the string is just the experiment name, do you need this string at all exposed for localization?

meandavejustice replied:
good point, I can just remove this, thanks

Commit:
Merge branch 'featured-experiments' of https://github.com/mozilla/testpilot into featured-experiments

johngruen committed:
fixall the styles (#3155)

johngruen committed:
fixall the styles (#3155)

lmorchard committed:
Merge branch 'l10n'

Commit:
Merge branch 'featured-experiments' of https://github.com/mozilla/testpilot into featured-experiments
